dcore/advection.py

```diff
@@ -122,3 +115,53 @@ def apply(self, x_in, x_out):
         dt = self.state.timestepping.dt
         self.solver.solve()
         x_out.assign(x_in + dt*self.dq)
+
+
+class DGAdvection(Advection):
+
+    def __init__(self, state):
+
+        super(DGAdvection, self).__init__(state)
+
+        dt = state.timestepping.dt
+
+        # by convention the last space is the DG space
+        DGSpace = state.V[-1]
```

Comment:
>> +class DGAdvection(Advection):
>> +
>> +    def __init__(self, state):
>> +
>> +        super(DGAdvection, self).__init__(state)
>> +
>> +        dt = state.timestepping.dt
>> +
>> +        # by convention the last space is the DG space
>> +        DGSpace = state.V[-1]


>Could you not just have a list and put it in Form order. That would provide a definitive reason >why DG is space -1, and would provide places for the other ones?

Yes, this was my reasoning - but I didn't realise Colin had dome something different for the 3D code.